services/explorer-indexer/src/modules/historical-sync/historical.syncer.service.ts

```diff
@@ -219,21 +219,28 @@ export default class HistoricalSyncService implements OnModuleInit {
       this.syncState.number < this.chainState.number;
       this.syncState.number++
     ) {
-      chuncks.push({
+      if (chunks.length === 1000) {
+        try {
+          const added = await this.pbftsQueue.addBulk(chunks);
+          if (added) {
+            this.logger.log(
+              `Added ${chunks.length} PBFT periods in Queue up until ${
+                chunks[chunks.length - 1].data.pbftPeriod
+              }`
+            );
+            chunks.length = 0;
```

Comment:
Whaaa' ? :D

Reply:
The problem was that there were too many jobs created all at once, that's why the heap got full, so now we create chunks of 1000, send it bulk to the queue and then clear the array. 😄